app/src/main/java/uw/hcrlab/kubi/lesson/results/NameResult.java

```diff
@@ -1,12 +1,40 @@
 package uw.hcrlab.kubi.lesson.results;
 
+import java.util.ArrayList;
+
 import uw.hcrlab.kubi.lesson.Result;
 
 /**
  * Created by Alexander on 4/25/2016.
  */
 public class NameResult extends Result {
-    public NameResult(boolean isCorrect) {
-        super(isCorrect);
+    private String blame;
+
+    private ArrayList<String> solutions;
+
+    public NameResult(boolean correct) {
+        super(correct);
+
+        solutions = new ArrayList<>();
+    }
+
+    public void setBlame(String blame) {
+        this.blame = blame;
+    }
+
+    public String getBlame() {
+        return blame;
+    }
+
+    public boolean hasBlame() {
+        return blame != null;
+    }
```

Comment:
I was hesitant to since not all results can have a list of solutions or blame text. It just happens that TRANSLATE and NAME do. I wrote them separate so that they are customized to the specific data structure I am putting in Firebase, and independent of other types of results.